python/cuml/preprocessing/TargetEncoder.py

```diff
@@ -37,9 +37,8 @@ class TargetEncoder:
         Default number of folds for fitting training data. To prevent
         label leakage in `fit`, we split data into `n_folds` and
         encode one fold using the target variables of the remaining folds.
-    smooth : float (default=0)
-        0 <= smooth <= 1
-        Percentage of samples to smooth the encoding
+    smooth : int (default=0)
+        Count of samples to smooth the encoding. 0 means no smoothing.
```

Comment:
We should add a deprecation warning for using floats for one version since this is a public API breaking change. I wonder if perhaps it would be a good idea to accept either a float or int and behave accordingly as opposed to change to only accept int?

Reply:
Good idea. Float will work just fine.